### ISSUE TITLE
Pattern locals declared in initializers of 'using' locals cannot be referenced

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#9608]
```
        [Fact]
        public void Test()
        {
            var source =
@"
public class X
{
    public static void Main()
    {
        using (System.IDisposable d1 = Dummy(new object() is var x1,
                                             x1))
        {
            System.Console.WriteLine(x1);
        }
    }

    static System.IDisposable Dummy(params object[] y) {return null;}
}
";
            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
            compilation.VerifyDiagnostics(
                );
        }
```

The test fails with unexpected errors:
```
// (7,46): error CS0103: The name 'x1' does not exist in the current context
//                                              x1))
Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(7, 46),
// (9,38): error CS0103: The name 'x1' does not exist in the current context
//             System.Console.WriteLine(x1);
Diagnostic(ErrorCode.ERR_NameNotInContext, "x1").WithArguments("x1").WithLocation(9, 38)
```

**Comment by AlekseyTs** [comment]
Related to #8817
